mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/massindexing/impl/PojoMassIndexingBatchCoordinator.java

```diff
@@ -96,12 +92,10 @@ public void runWithFailureHandler() throws InterruptedException {
 	 * Operations to do before the multiple-threads start indexing
 	 */
 	private void beforeBatch() throws InterruptedException {
-		// Create an agent to suspend concurrent indexing
-		agent = mappingContext.createMassIndexerAgent(
-				new PojoMassIndexerAgentCreateContextImpl( mappingContext, detachedSession.tenantIdentifier() )
-		);
-		// Start the agent and wait until concurrent indexing actually gets suspended
-		Futures.unwrappedExceptionGet( agent.start() );
+		for ( BatchCoordinatorContext context : coordinatorContexts ) {
+			// Start the agent and wait until concurrent indexing actually gets suspended
+			Futures.unwrappedExceptionGet( context.agent().start() );
+		}
```

Comment:
It would be better to do this in parallel. See `org.hibernate.search.mapper.pojo.work.impl.PojoScopeWorkspaceImpl#doOperationOnTypes` for how to do that properly (basically create an array of futures then use `Futures.unwrappedExceptionGet( CompletableFuture.allOf( futures ) )`).